src/client/app/components/unit/EditUnitModalComponent.tsx

```diff
@@ -156,62 +158,125 @@ export default function EditUnitModalComponent(props: EditUnitModalComponentProp
-		for (const value of Object.values(meterDataByID)) {
-			// This unit is used by a meter so cannot be deleted. Note if in a group then in a meter so covers both.
-			if (value.unitId === state.id) {
-				// TODO see EditMeterModalComponent for issue with line breaks. Same issue in strings below.
-				error_message += ` ${translate('meter')} "${value.name}" ${translate('uses')} ${translate('unit')} ` +
-					`"${state.name}" ${translate('as.meter.unit')};`;
-			}
-			if (value.defaultGraphicUnit === state.id) {
-				error_message += ` ${translate('meter')} "${value.name}" ${translate('uses')} ${translate('unit')} ` +
-					`"${state.name}" ${translate('as.meter.defaultgraphicunit')};`;
-			}
-		}
+		// Check conversions first (highest priority)
+		const conversionSources: React.ReactNode[] = [];
+		const conversionDestinations: React.ReactNode[] = [];
 		for (let i = 0; i < conversionData.length; i++) {
 			if (conversionData[i].sourceId === state.id) {
 				// This unit is the source of a conversion so cannot be deleted.
-				error_message += ` ${translate('conversion')} ${unitDataByID[conversionData[i].sourceId].name}` +
-					`${conversionArrow(conversionData[i].bidirectional)}` +
-					`${unitDataByID[conversionData[i].destinationId].name} ${translate('uses')} ${translate('unit')}` +
-					` "${state.name}" ${translate('unit.source.error')};`;
+				conversionSources.push(
+					<div key={`conversion-source-${i}`}>
+						{translate('conversion')} {unitDataByID[conversionData[i].sourceId].name}
+						{conversionArrow(conversionData[i].bidirectional)}
+						{unitDataByID[conversionData[i].destinationId].name} {translate('uses')} {translate('unit')}"{state.name}" {translate('unit.source.error')}
+					</div>
+				);
+				cancel = true;
 			}
 
 			if (conversionData[i].destinationId === state.id) {
 				// This unit is the destination of a conversion so cannot be deleted.
-				error_message += ` ${translate('conversion')} ${unitDataByID[conversionData[i].sourceId].name}` +
-					`${conversionArrow(conversionData[i].bidirectional)}` +
-					`${unitDataByID[conversionData[i].destinationId].name} ${translate('uses')} ${translate('unit')}` +
-					` "${state.name}" ${translate('unit.destination.error')};`;
+				conversionDestinations.push(
+					<div key={`conversion-dest-${i}`}>
+						{translate('conversion')} {unitDataByID[conversionData[i].sourceId].name}
+						{conversionArrow(conversionData[i].bidirectional)}
+						{unitDataByID[conversionData[i].destinationId].name} {translate('uses')} {translate('unit')}"{state.name}" {translate('unit.destination.error')}
+					</div>
+				);
+				cancel = true;
+			}
+		}
+
+		// Check meters using this unit
+		const metersUsingUnit = Object.values(meterDataByID).filter(meter => meter.unitId === state.id);
+
+		if (cancel || metersUsingUnit.length > 0) {
+			if (conversionSources.length > 0 || conversionDestinations.length > 0 || metersUsingUnit.length > 0) {
+				msgElements.push(
```

Comment:
I was thinking that creating conversionSources, conversionDestinations & metersUsingUnit to then push into msgElements seems like extra work. Would it be possible to check if any happened and then add this message as a new index 0 in msgElements array (maybe with splice)?